### PR TITLE
Add tgaff implicit wait tests

### PR DESCRIPTION
**Pull request by luke-hill**
This PR is aimed at doing 3 things.
- Adding a bunch of well written tests from @tgaff - Which have been shamelessly stolen, and very slightly adapted
- Tweaking a little bit of documentation, as well as the run order of the Travis jobs (Again thanks to @tgaff)
- Implicitly (Badoom-tish), show that actually in the Capybara3 rework branch I worked on. I've fixed the implicit waiting issue. The tests shown here were designed against Travis' code, but actually work fine against both master and the code which this is based off.

Note that due to Point 3 - This item requires #277 to be merged in first (Which will then remove the first 11 commits), at which point a review can be undertaken on the new 5 commits here.

TL;DR - After the code contained in this full PR. The next gem can be released which fixes a couple of bugs, clears some tech debt, and enables Capybara 3 support.

### CODE REVIEW

**Comment by luke-hill** [comment]
@tgaff Can you review / check you're ok with this. @ineverov also if you have capacity

**Comment by luke-hill** [comment]
Ignore the :red_circle: - One timeout on one of the tests.

**Comment by tgaff** [comment]
@luke-hill Thanks for getting this in.  And sorry, I've been too overwhelmed to keep up with stuff.

**Comment by luke-hill** [comment]
No probs. `2.15` introduced Capybara3 capabilities (Limited), and fixed the implicit waiting.
`2.15.1` fixed a load of dev side code, 1 major bug (iFrame creation inside sections), and upped some dependencies - Max capybara is now `< 3.3`
